backend/src/users/user.service.ts

```diff
@@ -22,14 +22,31 @@ export class UsersService {
         return this.userModel.find().exec();
     }
 
-    async findOne(@Param(':id') id: number | string) {
+    async findById(@Param(':id') id: number | string) {
         const users = await this.userModel.find().exec()
         const user = users.filter(user => user._id.toString() === id)
         return user
     }
 
+    async findOne(@Param() params) {
+        if (!params.password || !params.email) {
+            return null
+        }
+
+        try {
+            const user = await this.userModel.findOne({email: params.email})
+            const compare = await bcrypt.compare(params.password, user.password)
+
+            if(compare) {
+                return user.token
+            }
```

Comment:
What happens when `compare` is false?
It should throw an error probably, like `Username / password combination is incorrect`.